pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -122,6 +123,9 @@
     private final SubscriptionMode subscriptionMode;
     private volatile BatchMessageIdImpl startMessageId;
 
+    private volatile BatchMessageIdImpl seekMessageId;
+    private final AtomicBoolean duringSeek;
```

Comment:
Is it possible only add `seekMessageId` to achieve the seek logic?

Reply:
I don't think so. Without the flag, we cannot tell if seekMessageId or lastDequeueMessage should be used as the start message id.